src/main/java/org/reaktivity/nukleus/kafka/internal/stream/NetworkConnectionPool.java

```diff
@@ -1661,18 +1662,22 @@ private void handleDescribeConfigsResponse(
             }
             if (error.isRecoverable())
             {
-                pendingTopicMetadata.reset();
+                pendingTopicMetadata.invalidate();
                 doRequestIfNeeded();
             }
             else
             {
-                TopicMetadata metadata = pendingTopicMetadata;
-                pendingTopicMetadata = null;
-                metadata.setErrorCode(error);
-                metadata.setCompacted(compacted);
-                metadata.setDeleteRetentionMs(deleteRetentionMs);
-                metadata.setComplete(true);
-                metadata.flush();
+                // Guard against a possible race with invalidateBroker called due to fetch connection failure
+                if (pendingTopicMetadata.nextRequiredRequestType == MetadataRequestType.DESCRIBE_CONFIGS)
+                {
+                    TopicMetadata metadata = pendingTopicMetadata;
+                    pendingTopicMetadata = null;
```

Comment:
Can this lead to NPE as `pendingTopicMetadata` is being dereferenced in a few places?

Reply:
I think it is safe. The setting of pendingTopicMetadata was already being done before my changes. All I did was to add the if statement.  pendingTopicMetadata is only dereferenced  while processing metadata responses, and while building requests. Since we don't do pipelined requests,  there is no possibility of subsequent response processing happening before we next build a request. And when we build a request we always guarantee that pendingTopicMetadata  is set (in  MetadataConnection.doRequestIfNeeded).